modin/pandas/dataframe.py

```diff
@@ -2074,17 +2074,20 @@ def squeeze(
         Squeeze 1 dimensional axis objects into scalars.
         """
         axis = self._get_axis_number(axis) if axis is not None else None
-        if axis is None and (len(self.columns) == 1 or len(self.index) == 1):
-            return Series(query_compiler=self._query_compiler).squeeze()
-        if axis == 1 and len(self.columns) == 1:
+        len_columns = len(self.columns)
+        if axis == 1 and len_columns == 1:
             self._query_compiler._shape_hint = "column"
             return Series(query_compiler=self._query_compiler)
-        if axis == 0 and len(self.index) == 1:
-            qc = self.T._query_compiler
-            qc._shape_hint = "column"
-            return Series(query_compiler=qc)
-        else:
-            return self.copy()
+        if axis in [0, None]:
```

Comment:
At first glance, it seems like this is just a refactoring. @noloerino сould you please highlight the place where the logic changes?

Reply:
I would still like to merge this pull request as a refactoring. We just need to change its name.

Reply:
I changed the title to indicate a refactor.

That said, I'm not sure anymore if this refactor makes sense. In the new code, it's possible for `len_columns` to be computed unnecessarily when `axis=0`, and `len_index` to be computed when `axis is None and len_columns == 1`. Inlining those variable references basically puts us back to what the original code was.

Reply:
I modified the PR to use `len(self)` instead of `len(self.index)`, which could in some cases avoid materializing the index. This is a subset of #7398's changes though, so I'm fine with closing this and getting that merged instead.

Reply:
It's fine, let's merge it :)